File: src/Go/pre-go/encode.go
package main
import "strings"
func (m mode) encode(x string) (X ,y string, g int) {
	for _,h := range m {
		if strings.HasPrefix(x,h.reads) {
			X = x[len(h.reads):]
			y = h.writes
			g = h.goes
			return
		}
	}
	return
}
func (f filter) encode(x string) (string) {
	current_mode := 0
	y := ""
	z := ""
	for ;x != ""; {
		x,z, current_mode = f[current_mode].encode(x)
		y = y + z
	} 
	return y
}

File: src/Julia/prefix-julia-plus/encode.jl
function encode(p :: Word , q :: Key) :: Word
    k = deepcopy(q)
    c :: Word = []
    m = 0
    i = 1
    while i <= length(p)
        for r in k
            if r.mode == m && prefix(r.reads,p[i:end])
                #push!(c,r.writes)
                c = vcat(c, r.writes)
                i += length(r.reads)
                m = r.goes
                @goto next_mode
            end
        end
        @label next_mode
    end
    c
end